yt_dlp/extractor/tver.py

```diff
@@ -1,77 +1,94 @@
 # coding: utf-8
 from __future__ import unicode_literals
 
-
 from .common import InfoExtractor
-from ..compat import compat_str
 from ..utils import (
     ExtractorError,
-    int_or_none,
-    remove_start,
     smuggle_url,
+    str_or_none,
     traverse_obj,
 )
 
 
 class TVerIE(InfoExtractor):
-    _VALID_URL = r'https?://(?:www\.)?tver\.jp/(?P<path>corner|episode|feature|lp|tokyo2020/video)/(?P<id>[fc]?\d+)'
-    # videos are only available for 7 days
+    _VALID_URL = r'https?://(?:www\.)?tver\.jp/(?:(?P<type>lp|corner|series|episodes?|feature|tokyo2020/video)/)+(?P<id>[a-zA-Z0-9]+)'
     _TESTS = [{
-        'url': 'https://tver.jp/corner/f0062178',
-        'only_matching': True,
+        'skip': 'videos are only available for 7 days',
+        'url': 'https://tver.jp/episodes/ephss8yveb',
+        'info_dict': {
+            'title': '#44　料理と値段と店主にびっくり　オモてなしすぎウマい店　2時間SP',
+            'description': 'md5:4ae6bb0d8e9c860bf91bdb791e32356e8bd2d657a588e154f69b2cf6c6cbaf48',
+        },
+        'add_ie': ['BrightcoveNew'],
     }, {
-        'url': 'https://tver.jp/feature/f0062413',
-        'only_matching': True,
-    }, {
-        'url': 'https://tver.jp/episode/79622438',
-        'only_matching': True,
-    }, {
-        # subtitle = ' '
-        'url': 'https://tver.jp/corner/f0068870',
-        'only_matching': True,
-    }, {
-        'url': 'https://tver.jp/lp/f0009694',
-        'only_matching': True,
+        'skip': 'videos are only available for 7 days',
+        'url': 'https://tver.jp/lp/episodes/ep6f16g26p',
+        'info_dict': {
+            # sorry but this is "correct"
+            'title': '4月11日(月)23時06分 ~ 放送予定',
+            'description': 'md5:c8ed3a7507e9a087a922b144e79b93336da1b60ca1d867df438ee63e7b7093dc',
+        },
+        'add_ie': ['BrightcoveNew'],
     }, {
-        'url': 'https://tver.jp/lp/c0000239',
+        'url': 'https://tver.jp/corner/f0103888',
         'only_matching': True,
     }, {
-        'url': 'https://tver.jp/tokyo2020/video/6264525510001',
+        'url': 'https://tver.jp/lp/f0033031',
         'only_matching': True,
     }]
-    _TOKEN = None
     BRIGHTCOVE_URL_TEMPLATE = 'http://players.brightcove.net/%s/default_default/index.html?videoId=%s'
```

Comment:
Although this isn't part of the PR, isn't the ability to populate `BRIGHTCOVE_URL_TEMPLATE` something that should be a method or variable of the `BrightCoveNewIE` class and imported from there?

Reply:
:shrug:

Reply:
I agree that is good, but it will need (minor) refactoring across multiple extractors